### PR TITLE
Feature/roulette minigame

### LINKED ISSUE #70: Minispiel: Roulette

### PR DESCRIPTION
**Pull request by soni4295**
fixes #70 roulette mini game ui and networkmessages

### CODE REVIEW

**Review by orazefabian — DISMISSED** [review]
schaut nice aus 👍 
aber könntest vlt probieren a paar von de 63 code smells zu entfernen

File: core/src/main/java/com/mankomania/game/core/network/messages/clienttoserver/roulette/RouletteStakeMessage.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
+package com.mankomania.game.core.network.messages.clienttoserver.roulette;
+
+public class RouletteStakeMessage {
+    //message for server
+    private int rsmPlayerId;
```
Comment (Th3Dilli) [inline core/src/main/java/com/mankomania/game/core/network/messages/clienttoserver/roulette/RouletteStakeMessage.java]: rename to rsmPlayerIndex to be more precise

File: core/src/main/java/com/mankomania/game/core/network/messages/clienttoserver/roulette/StartRouletteClient.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
+package com.mankomania.game.core.network.messages.clienttoserver.roulette;
+
+public class StartRouletteClient {
+    //Message verschickt, wenn ein Spieler auf das Rouletteminispielfeld kommt
+    private int playerId;
```
Comment (Th3Dilli) [inline core/src/main/java/com/mankomania/game/core/network/messages/clienttoserver/roulette/StartRouletteClient.java]: rename to playerIndex to be more precise

File: core/src/main/java/com/mankomania/game/core/network/messages/servertoclient/roulette/RouletteResultMessage.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
+package com.mankomania.game.core.network.messages.servertoclient.roulette;
+
+public class RouletteResultMessage {
+    private int playerID;
```
Comment (Th3Dilli) [inline core/src/main/java/com/mankomania/game/core/network/messages/servertoclient/roulette/RouletteResultMessage.java]: rename to playerIndex to be more precise

File: core/src/main/java/com/mankomania/game/core/network/messages/servertoclient/roulette/StartRouletteServer.java [hunk 4]
```diff
@@ -0,0 +1,21 @@
+package com.mankomania.game.core.network.messages.servertoclient.roulette;
+
+public class StartRouletteServer {
+    //Nachricht von Server an Clients, dass das Rouletteminigame startet
+    private int playerId;
```
Comment (Th3Dilli) [inline core/src/main/java/com/mankomania/game/core/network/messages/servertoclient/roulette/StartRouletteServer.java]: rename to playerIndex to be more precise

File: server/src/main/java/com/mankomania/game/server/minigames/RouletteHandler.java [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+
+public class RouletteHandler {
+    private ServerData serverData;
+    private Server server;
+    private ArrayList<RouletteStakeMessage> inputPlayerBets;
+    private static final String[] COLOR = {"red", "black", "red", "black", "red", "black", "red", "black", "red", "black", "red", "black", "red", "black", "red", "black", "red", "black", "red", "black", "red", "black", "red", "black", "red", "black", "red", "black", "red", "black", "red", "black", "red", "black", "red", "black"};
```
Comment (Th3Dilli) [inline server/src/main/java/com/mankomania/game/server/minigames/RouletteHandler.java]: use modulo (%2) to when trying to get color

File: gamecore/src/main/java/com/mankomania/game/gamecore/screens/RouletteMiniGameScreen.java [hunk 6]
```diff
@@ -0,0 +1,416 @@
...
+                textButton5.setColor(Color.RED);
+                textFieldInputPlayer.setText(BLACK);
+            }
+        });
+        stage.addActor(tableMain);
+    }
+
+    public boolean isNumeric(String strNum) {
+
+        if ((strNum.matches("[0-9]+") && strNum.length() <= 2)) {
```
Comment (Th3Dilli) [inline gamecore/src/main/java/com/mankomania/game/gamecore/screens/RouletteMiniGameScreen.java]: return the condition instead

**Review by Th3Dilli — APPROVED** [review]

**Comment by soni4295** [comment]
#70

**Pull request merged**